contracts/RiskSteward/MarketCapsRiskSteward.sol

```diff
@@ -0,0 +1,198 @@
+            _processBorrowCapUpdate(update);
+        } else {
+            revert UnsupportedUpdateType();
+        }
+    }
+
+    function _updateSupplyCaps(address market, bytes memory newValue) internal {
+        address comptroller = IVToken(market).comptroller();
+        address[] memory newSupplyCapMarkets = new address[](1);
+        newSupplyCapMarkets[0] = market;
+        uint256[] memory newSupplyCaps = new uint256[](1);
+
+        newSupplyCaps[0] = uint256(bytes32(newValue));
+        if (comptroller == address(CORE_POOL_COMPTROLLER)) {
+            ICorePoolComptroller(comptroller)._setMarketSupplyCaps(newSupplyCapMarkets, newSupplyCaps);
+        } else {
+            IIsolatedPoolsComptroller(comptroller).setMarketSupplyCaps(newSupplyCapMarkets, newSupplyCaps);
+        }
+        emit SupplyCapUpdated(market, newSupplyCaps[0]);
+    }
+
+    function _updateBorrowCaps(address market, bytes memory newValue) internal {
+        address comptroller = IVToken(market).comptroller();
+        address[] memory newBorrowCapMarkets = new address[](1);
+        newBorrowCapMarkets[0] = market;
+        uint256[] memory newBorrowCaps = new uint256[](1);
+        newBorrowCaps[0] = uint256(bytes32(newValue));
+        if (comptroller == address(CORE_POOL_COMPTROLLER)) {
+            ICorePoolComptroller(comptroller)._setMarketBorrowCaps(newBorrowCapMarkets, newBorrowCaps);
+        } else {
+            IIsolatedPoolsComptroller(comptroller).setMarketBorrowCaps(newBorrowCapMarkets, newBorrowCaps);
+        }
+        emit BorrowCapUpdated(market, newBorrowCaps[0]);
+    }
+
+    function _processSupplyCapUpdate(RiskParameterUpdate memory update) internal {
+        _validateSupplyCapUpdate(update);
+        _updateSupplyCaps(update.market, update.newValue);
+    }
+
+    function _processBorrowCapUpdate(RiskParameterUpdate memory update) internal {
+        _validateBorrowCapUpdate(update);
+        _updateBorrowCaps(update.market, update.newValue);
+    }
+
+    function _validateSupplyCapUpdate(RiskParameterUpdate memory update) internal view {
+        ICorePoolComptroller comptroller = ICorePoolComptroller(IVToken(update.market).comptroller());
+        uint256 currentSupplyCap = comptroller.supplyCaps(address(update.market));
+
+        uint256 newValue = abi.decode(
+            abi.encodePacked(new bytes(32 - update.newValue.length), update.newValue),
+            (uint256)
+        );
+        _updateWithinAllowedRange(currentSupplyCap, newValue);
+    }
+
+    function _validateBorrowCapUpdate(RiskParameterUpdate memory update) internal view {
+        ICorePoolComptroller comptroller = ICorePoolComptroller(IVToken(update.market).comptroller());
+        uint256 currentBorrowCap = comptroller.borrowCaps(address(update.market));
+
+        uint256 newValue = abi.decode(
+            abi.encodePacked(new bytes(32 - update.newValue.length), update.newValue),
+            (uint256)
+        );
```

Comment:
Would it make sense to extract this into an internal function? We are decoding this value in several places...

Reply:
I updated this but it must not have been pushed